### DISCUSSION TITLE
Terminal UI API similar to `spectre.console`- is it planned?

### DISCUSSION THREAD

**Discussion by sandreas** [Tyrrrz/CliFx#130] [Q&A] [answered]
Hello,

after using `CliFX` and enjoying the good code quality and your developer feedback I switched to [`spectre.console`](https://spectreconsole.net/) (no offense) for my Project [`tone`](https://github.com/sandreas/tone), because it is a bit nicer to look at :-) 

Recently I had some nasty issues in very basic features (e.g. Argument parsing) that made me a bit nervous and I thought about coming back to `CliFX`... 

Now I just wanted to ask, if you plan to provide a "nicer look" / "Terminal UI" API, that might be similar or even better than `spectre.console`?

Thank you.

**Comment by Tyrrrz** [comment]
You can just combine the two: https://github.com/Tyrrrz/CliFx/wiki/Integrating-with-Spectre.Console

**Comment by sandreas** [comment, marked as answer]
> You can just combine the two: https://github.com/Tyrrrz/CliFx/wiki/Integrating-with-Spectre.Console

Yes, I know. But ATM I'm not convinced that I'm happy with `spectre.console`s way of dealing with issues... You responded in 2 hours... So I was hoping that maybe there is a plan to deploy your own UI thingy... :-) But as you said, maybe this would be not worth the effort.

So I can take as response, that there is no such plan for a Terminal UI Widget API?

Thanks for the quick response.

**Reply by Tyrrrz** [reply]
You don't have to use spectre's argument parsing (which is what I understand you're having issues with?). You'd be using spectre only for UI.

No, there is no plan to implement anything similar. Spectre does a decent job as it is and is easy to integrate, so I don't see a reason to reinvent the wheel.

**Reply by sandreas** [reply]
> which is what I understand you're having issues with?

This is at least one part of my problem :-) But you are right, don't reinvent the wheel... thank you.
